adoc/code/graph.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2025 The Khronos Group, Inc.
+// SPDX-License-Identifier: MIT
+
+#include <iostream>
+#include <sycl/sycl.hpp>
+#include <utility>
+using namespace sycl;  // (optional) avoids need for "sycl::" before SYCL names
+
+constexpr size_t N = 100;
+constexpr size_t Iter = 10;
+
+int main() {
+  // Create a queue to work on.
+  queue myQueue({property::queue::in_order{}});
+
+  // Create some 1D arrays of float
+  float* AOld = malloc_device<float>(N, myQueue);
+  float* ANew = malloc_device<float>(N, myQueue);
+
+  khr::command_graph myGraph{myQueue};
+
+  myGraph.begin_recording(myQueue);
+
+  // Unroll iterative 1D solver by a factor of two to enable
+  // recording into an immutable graph.
+  //
+  // Record an asynchronous kernel to compute even timesteps
+  myQueue.submit([&](handler& cgh) {
+    // Enqueue a parallel kernel iterating on a N 1D iteration space
+    cgh.parallel_for(range<1>{N - 2}, [=](id<1> index) {
+      ANew[index + 1] = AOld[index] / 2 + AOld[index + 2] / 2;
+    });
+  });
+
+  std::swap(AOld, ANew);
```

Comment:
Do we need to call `std::swap` here or could we just swap the parameters around in the kernel?

Reply:
Possible, if we keep the two kernels explicitly and not use a single function as earlier suggested.
My preference is to keep the swap, just to make it more obvious. 
No objections on the single function to simplify the example.

Reply:
Actually now that I see this with the kernel function refactor, I wonder if this is doing what we intend, the lambda function `step` will capture the pointers by value, so calling `std::swap` will only swap those local copies, it doesn't change the lambda, so both kernel invocations will have the same input/output.

If we are to refactor the kernel, I think we'd need to create a new lambda object each time, but move that to a function which takes the pointer parameters. Alternatively we could keep the separate kernel definitions, in which case we could just swap the pointer parameters.